src/status_im2/common/standard_auth/view.cljs

```diff
@@ -0,0 +1,154 @@
+            [reagent.core :as reagent]
+            [quo2.foundations.colors :as colors]
+            [react-native.fast-image :as fast-image]
+            [status-im2.common.resources :as resources]
+            [utils.i18n :as i18n]
+            [quo2.components.keycard.view :as keycard]
+            [utils.re-frame :as rf]))
+
+(defn enter-keycard-pin-sheet
+  []
+  (let [entered-numbers (reagent/atom [])
+        max-digits 6
+        max-attempt-reached (reagent/atom false)]
+    (fn []
+      (defn handle-number-entry [number]
+        (when (>= (count @entered-numbers) max-digits)
+          (reset! entered-numbers []))
+        (when (= (count @entered-numbers) (dec max-digits))
+          (reset! max-attempt-reached true))
+        (swap! entered-numbers conj number))
+
+      (println "entered-numbers" @entered-numbers)
+
+      (defn get-dot-style [idx]
+        (if (<= idx (dec (count @entered-numbers)))
+          (if @max-attempt-reached
+            (assoc style/digit :backgroundColor colors/danger)
+            (assoc style/digit :backgroundColor colors/white))
+          (assoc style/digit :backgroundColor colors/neutral-50)))
+
+      [rn/view {:style style/container}
+       [rn/view {:style style/inner-container}
+        [quo/text {:accessibility-label :enter-keycard-pin-text :size :heading-2 :weight :semi-bold} "Enter Keycard PIN"]
+
+        [rn/view {:style style/context-tag}
+         [quo/context-tag
+          {:type           :icon
+           :icon           :i/placeholder
+           :size            24
+           :context     "Alisher Card"}]]
+
+        [rn/view
+         (if @max-attempt-reached
+           {:style style/digits-container :margin-bottom 0}
+           {:style style/digits-container :margin-bottom 34})
+         (for [i (range max-digits)]
+           [rn/view
+            {:key i
+             :style (get-dot-style i)}])]
+
+        (when @max-attempt-reached
+          [rn/view {:style style/max-attempt-reached-container}
+           [quo/text {:size :label :style {:color colors/danger}} "4 attempts left"]])]
+
+       [numbered-keyboard/view
+        {:disabled?   false
+         :on-press    handle-number-entry
+         :blur?       false
+         :delete-key? false
+         :left-action :none}]])))
+
+(defn this-is-not-keycard-sheet
+  []
+  [rn/view {:style style/container}
+   [rn/view {:style style/inner-container}
+    [quo/button
+     {:container-style style/close-button
+      :type :grey
+      :icon-only? true
+      :size 32
+      :on-press        #(rf/dispatch [:hide-bottom-sheet])} :i/close]
+
+    [quo/text
+     {:accessibility-label :this-is-not-keycard-text :size :heading-1 :weight :semi-bold} "Oops, this isn’t a Keycard"]
+
+    [quo/text {:accessibility-label :make-sure-text :size :paragraph-1 :weight :medium :style style/secondary-text} "Make sure the card you scanned is a Keycard."]]
+
+   [fast-image/fast-image
+    {:style  {:width  "100%"
```

Comment:
How about trying flexbox properties instead of `:width "100%"`? Our [guidelines](https://github.com/status-im/status-mobile/blob/develop/doc/new-guidelines.md#dont-use-percents-to-define-widthheight) recommend it.

Reply:
@ajayesivan I missed the guideline for this one. Now it's updated.